### PR TITLE
whatsapp-for-linux: init-at 1.1.5

### PR DESCRIPTION
**Pull request by wandersoncferreira**
Whataspp is widely used in several contries as default messaging
system. We can rely on browser-based solution but an app is preferred
method for several users.

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by wandersoncferreira** [comment]
Result of `nixpkgs-review pr 117908` [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>whatsapp-for-linux</li>
  </ul>
</details>

File: pkgs/applications/networking/instant-messengers/whatsapp-for-linux/default.nix [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+  runHook preInstall
+
+  cmake -DCMAKE_BUILD_TYPE=Release .
+  make
+  make install
+  runHook postInstall
+  '';
+
+  meta = with lib; {
+    homepage = "https://www.whatsapp.com/";
```
Comment (doronbehar) [inline pkgs/applications/networking/instant-messengers/whatsapp-for-linux/default.nix:39]:
```suggestion
    homepage = "https://github.com/eneshecan/whatsapp-for-linux";
```

File: pkgs/applications/networking/instant-messengers/whatsapp-for-linux/default.nix [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+libappindicator-gtk3, gst_all_1, pcre }:
+
+stdenv.mkDerivation rec {
+  pname = "whatsapp-for-linux";
+  version = "1.1.5";
+
+  src = fetchurl {
+    url = "https://github.com/eneshecan/whatsapp-for-linux/archive/v${version}.tar.gz";
+    sha256 = "6e034d9f7d1dca94df943276ce91227088bc986442ee095d0904f68fb7abbf44";
+  };
```
Comment (doronbehar) [inline pkgs/applications/networking/instant-messengers/whatsapp-for-linux/default.nix:11]:
```suggestion
  src = fetchFromGitHub {
    owner = "eneshecan";
    repo = pname;
    rev = "v${version}";
    sha256 = "...";
  };
```

File: pkgs/applications/networking/instant-messengers/whatsapp-for-linux/default.nix [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+
+  nativeBuildInputs = [
+    cmake
+    pkg-config
+    gnome3.gtkmm
```
Comment (doronbehar) [inline pkgs/applications/networking/instant-messengers/whatsapp-for-linux/default.nix:16]: Most of the deps listed are not native:

```suggestion
  ];
  
  buildInputs = [
    gnome3.gtkmm
```

File: pkgs/applications/networking/instant-messengers/whatsapp-for-linux/default.nix [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+    libappindicator-gtk3
+    gst_all_1.gst-plugins-base
+    gst_all_1.gst-plugins-good
+    gst_all_1.gst-plugins-bad
+    gst_all_1.gst-libav
+    pcre
+  ];
+
+  dontStrip = true;
+  dontPatchELF = true;
```
Comment (doronbehar) [inline pkgs/applications/networking/instant-messengers/whatsapp-for-linux/default.nix:27]: Why?

File: pkgs/applications/networking/instant-messengers/whatsapp-for-linux/default.nix [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+
+  installPhase = ''
+  runHook preInstall
+
+  cmake -DCMAKE_BUILD_TYPE=Release .
+  make
+  make install
+  runHook postInstall
+  '';
```
Comment (doronbehar) [inline pkgs/applications/networking/instant-messengers/whatsapp-for-linux/default.nix:36]: Letting stdenv do it doesn't work?

**Review by doronbehar — CHANGES_REQUESTED** [review]
Your PR should have eventually two commits:

```
maintainers: add <me>
<package-name>: init at <version>
```

File: pkgs/applications/networking/instant-messengers/whatsapp-for-linux/default.nix (hunk 1 above)
Comment (wandersoncferreira) [reply pkgs/applications/networking/instant-messengers/whatsapp-for-linux/default.nix:39]: Thanks for suggestion!

File: pkgs/applications/networking/instant-messengers/whatsapp-for-linux/default.nix (hunk 4 above)
Comment (SuperSandro2000) [reply pkgs/applications/networking/instant-messengers/whatsapp-for-linux/default.nix:27]: Stripping should only be turned of if it breaks the binary.

**Comment by wandersoncferreira** [comment]
> Your PR should have eventually two commits:
> 
> ```
> maintainers: add <me>
> <package-name>: init at <version>
> ```

Thanks for the feedbacks, this is my first contribution. I will test the options mentioned again. For the commit message, the appropriate thing to do is to rebase the changes into these 2 commits, right? thanks

File: pkgs/applications/networking/instant-messengers/whatsapp-for-linux/default.nix (hunk 4 above)
Comment (wandersoncferreira) [reply pkgs/applications/networking/instant-messengers/whatsapp-for-linux/default.nix:32]: Will make another test on my side.

**Comment by doronbehar** [comment]
> For the commit message, the appropriate thing to do is to rebase the changes into these 2 commits, right? thanks

Yes, I don't know if there's an easy way in git to split commits, but I would roughly do this, in the current situation:

```
git clean # if needed
git co master
git co -B whatsapp-for-linux
curl https://github.com/NixOS/nixpkgs/pull/117908.patch | git apply
git add lib/maintainers.nix # I think this path is correct
git cmt -m "maintainers: add <me>"
git add . # add the package
git cmt -m "whatsapp-for-linux: init at <version>"
```

Afterwards, you can edit the expression to actually make it build and run, and everytime you need to change something and you need to commit and push it, do:

```
git commit --amend --no-edit
git push -f # Use the remote name if needed, or use -u to set it once for the remote of your fork
```

**Comment by doronbehar** [comment]
Note that the `curl` command needs `-L`.

File: pkgs/applications/networking/instant-messengers/whatsapp-for-linux/default.nix (hunk 5 above)
Comment (wandersoncferreira) [reply pkgs/applications/networking/instant-messengers/whatsapp-for-linux/default.nix]: It works, I removed the custom installPhase

File: pkgs/applications/networking/instant-messengers/whatsapp-for-linux/default.nix (hunk 4 above)
Comment (wandersoncferreira) [reply pkgs/applications/networking/instant-messengers/whatsapp-for-linux/default.nix]: Do not break the binary. Thanks for pointing out. Removed from latest

**Comment by wandersoncferreira** [comment]
Result of `nixpkgs-review pr 117908` [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>whatsapp-for-linux</li>
  </ul>
</details>

**Review by doronbehar — APPROVED** [review]

**Pull request merged**